### ISSUE TITLE
Option to disable the instant reorientation when going through a portal

### ISSUE THREAD

**Issue by Zod1234** [Gistix/portal2vr#55]
I understand this was probably done to alleviate motion sickness, but for me it's a little disorientating and immersion breaking.  

Comfort options are great to allow more users to play but they should be just that: optional.

I'm not asking to make this the default, just give me the option via a config setting.

**Comment by Audisek** [comment]
I second this, the smooth transition is crucial to keep track of where I am flying. 

Without it I get lost a lot when I jump through portals and I need to look around to orientate myself. Especially due to the reduced FOV compared to playing on a monitor.

**Comment by VirgilCore** [comment]
Yes. Please let us experience the rotate

**Comment by Flux001337** [comment]
Yes, an option to disable the instant reorientation will be will increase the immersion and experience a lot.
Please add an option to the config file.

**Comment by ORanGeOfficial** [comment]
Yes, please. The lack of smooth transition is very disorienting

**Comment by imcommonsense** [comment]
as much as i would love this, i don't quite know how this would work in VR.
rotating on the x axis seems simple enough, but i think that the y axis would screw with the headset.

**Comment by ORanGeOfficial** [comment]
> as much as i would love this, i don't quite know how this would work in VR. rotating on the x axis seems simple enough, but i think that the y axis would screw with the headset.

Just like in any other VR roller coaster game. The motion of the cart is independent of head movement. So a quick interpolation of all axes between pos1 (before entering a portal) and pos2 (after exiting another portal) should work fine. I mean the same interpolation as in flat screen portal 2.

**Comment by imcommonsense** [comment]
> > as much as i would love this, i don't quite know how this would work in VR. rotating on the x axis seems simple enough, but i think that the y axis would screw with the headset.
> 
> Just like in any other VR roller coaster game. The motion of the cart is independent of head movement. So a quick interpolation of all axes between pos1 (before entering a portal) and pos2 (after exiting another portal) should work fine. I mean the same interpolation as in flat screen portal 2.

alright then, thanks for explaining it to me
im not always the smartest lmao

**Comment by Zod1234** [comment]
"I mean the same interpolation as in flat screen portal 2."

That's what I'm asking for.

**Comment by TheIdiot0** [comment]
I.... sixth? Yea, sixth. I sixth this, this would be a good feature.

**Comment by FirEmerald** [comment]
Add me to the list of people who would like this as an _OPTION_ (make it default off though for the people who do get vr sickness please!)

**Comment by aladvs** [comment]
+1

**Comment by Tr1NgleDev** [comment]
+1

**Comment by ORanGeOfficial** [comment]
Here is the solution, but I can't make it to work.
https://github.com/Gistix/portal2vr/pull/103

**Comment by ORanGeOfficial** [comment]
Here is the fix by [vittorioromeo](https://github.com/vittorioromeo) and [sd805](https://github.com/sd805)

[camUpdate.zip](https://github.com/Gistix/portal2vr/files/13212271/camUpdate.zip)

Overwrite `d3d9.dll` in the `Portal 2\bin` folder and `config.txt` in the `Portal 2\VR` folder.
Also in the `config.txt` you can set `CameraUprightRecoverySpeed` to **0.03** to make the camera rotation speed closer to the original non VR game.

**Comment by Zod1234** [comment]
> Here is the fix by [vittorioromeo](https://github.com/vittorioromeo) and [sd805](https://github.com/sd805)
> 
> [camUpdate.zip](https://github.com/Gistix/portal2vr/files/13212271/camUpdate.zip)
> 
> Overwrite `d3d9.dll` in the `Portal 2\bin` folder and `config.txt` in the `Portal 2\VR` folder. Also in the `config.txt` you can set `CameraUprightRecoverySpeed` to **0.03** to make the camera rotation speed closer to the original non VR game.

this great.  only test a bit, but so far so good.  Will this become part of the main mod?
